jaxdecomp/_src/fft.py

```diff
@@ -75,46 +81,38 @@ def sfft_abstract_eval(x, fft_type, pdims, global_shape, adjoint):
     axis = 2
 
   output_shape = None
+  match fft_type:
+    case xla_client.FftType.FFT:
+      # FFT is X to Y to Z so Z-Pencil is returned
+      # Except if we are doing a YZ slab in which case we return a Y-Pencil
+      transpose_shape = (1, 2, 0) if not yz_slab else (2, 0, 1)
+      transposed_pdims = pdims if not yz_slab else pdims[::-1]
+    case xla_client.FftType.IFFT:
+      # IFFT is Z to X to Y so X-Pencil is returned
+      # In YZ slab case we only need one transposition back to get the X-Pencil
+      transpose_shape = (2, 0, 1) if not yz_slab else (1, 2, 0)
+      transposed_pdims = pdims
+    case _:
+      raise TypeError("only complex FFTs are currently supported through pfft.")
 
-  expected_slice_shape = (global_shape[0] // pdims[1],
-                          global_shape[1] // pdims[0], global_shape[2])
   # Are we operating on the global array?
   # This is called when the abstract_eval of the custom partitioning is called _custom_partitioning_abstract_eval in  https://github.com/google/jax/blob/main/jax/experimental/custom_partitioning.py#L223
   if x.shape == global_shape:
-    # only works for cubes
-    # TODO(wassim) The transpose has to be the same as the slices maybe?
-    output_shape = x.shape
+    shape = tuple([global_shape[i] for i in transpose_shape])
+    output_shape = shape
   # Or are we operating on a local slice?
   # this is called JAX calls make_jaxpr(lower_fn) in https://github.com/google/jax/blob/main/jax/experimental/custom_partitioning.py#L142C5-L142C35
-  elif x.shape == expected_slice_shape:
-
-    output_shape = expected_slice_shape
-
-    # why do this? there has to be an easier way to check validity of the shape
-    config = _jaxdecomp.GridConfig()
-
-    config.pdims = pdims
-    # Dimensions are actually in reverse order due to Fortran indexing at the cuDecomp level
-    config.gdims = global_shape[::-1]
-    config.halo_comm_backend = jaxdecomp.config.halo_comm_backend
-    config.transpose_comm_backend = jaxdecomp.config.transpose_comm_backend
-    # Dimensions are actually in reverse order due to Fortran indexing at the cuDecomp level
-    pencil = _jaxdecomp.get_pencil_info(config, axis)
```

Comment:
No need to ask cudecomp for their pencil info... I understood how they compute it and I reproduce here the same shape